data-prepper-plugins/saas-source-plugins/saas-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/saas_crawler/base/SaasSourcePlugin.java

```diff
@@ -0,0 +1,120 @@
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartitionStoreItem;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourceCoordinator;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourcePartition;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.UsesEnhancedSourceCoordination;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.PartitionFactory;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.partition.LeaderPartition;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.scheduler.LeaderScheduler;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.scheduler.WorkerScheduler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Function;
+
+
+
+/**
+ * JiraConnector connector entry point.
+ */
+
+public abstract class SaasSourcePlugin implements Source<Record<Event>>, UsesEnhancedSourceCoordination {
+
+
+  private static final Logger log = LoggerFactory.getLogger(SaasSourcePlugin.class);
+  private final PluginMetrics pluginMetrics;
+  private final PluginFactory pluginFactory;
+
+  private final AcknowledgementSetManager acknowledgementSetManager;
+
+  private final ExecutorService executorService;
+
+  private EnhancedSourceCoordinator coordinator;
+
+  @Getter
+  private final SaasSourceConfig sourceConfig;
+
+  private Buffer<Record<Event>> buffer;
+  private final Crawler crawler;
+
+
+  @DataPrepperPluginConstructor
+  public SaasSourcePlugin(final PluginMetrics pluginMetrics,
+                          final SaasSourceConfig sourceConfig,
+                          final PluginFactory pluginFactory,
+                          final AcknowledgementSetManager acknowledgementSetManager,
+                          final Crawler crawler,
+                          final SaasPluginExecutorServiceProvider executorServiceProvider) {
+    log.info("Create SaaS Source Connector");
+    this.pluginMetrics = pluginMetrics;
+    this.sourceConfig = sourceConfig;
+    this.pluginFactory = pluginFactory;
+    this.crawler = crawler;
+
+    this.acknowledgementSetManager = acknowledgementSetManager;
+    this.executorService = executorServiceProvider.get();
+  }
+
+
+  @Override
+  public void start(Buffer<Record<Event>> buffer) {
+    Objects.requireNonNull(coordinator);
+    log.info("Starting SaaS Source Plugin... ");
+    this.buffer = buffer;
+
+    boolean isPartitionCreated = coordinator.createPartition(new LeaderPartition());
+    log.info("Leader partition creation status: {}", isPartitionCreated);
```

Comment:
This will result in one of these logs whenever a new data prepper instance starts

```
Leader partition creation status: false
```
and isn't really helpful. Can be debug